commit 5414a2fa693356ed71d5f910e2ec1d5ca9bdf5bc (refs/changes/14/814/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-10-28T23:22:41+00:00 (2 years, 11 months ago)
    
    joyent/node-mname#7 names with trailing '.' encoded incorrectly
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/lib/dns-buffer.js b/lib/dns-buffer.js
index fbbc197..41c0cfd 100644
--- a/lib/dns-buffer.js
+++ b/lib/dns-buffer.js
@@ -143,6 +143,8 @@ DNSBuffer.prototype.writeName = function (name) {
                 rlen = part.length;
                 assert.ok(rlen < 64, 'segment "' + part + '" of name "' +
                     name + '" is too long');
+                if (rlen === 0 && rparts.length === 0)
+                        break;
                 this.writeUInt8(rlen);
                 this.write(new Buffer(part, 'ascii'));
         }
